index.html

```diff
@@ -482,6 +482,27 @@ <h3>
           Details of how to actually get the [=digital credential=] are
           forthcoming.
         </aside>
+        <ol class="algorithm" >
+          <li>
+            <aside class="issue">
+              The following guidance on handling multiple requests will need to be relocated once
+              this entire section of the algorithm is defined.
+            </aside>
+            When multiple [=digital credential/presentation requests=] are present
+            in {{DigitalCredentialRequestOptions/requests}},
+            [=user agents=] MUST only return one
+            [=digital credential/presentation responses | presentation response=] for one
+            of these requests.
+            Multiple credentials MAY be present in this one response depending on the capabilities
+            of the corresponding [=digital credential/exchange protocol=].
```

Comment:
This needs further refinement. I'm trying to capture the concept before getting the language just right. There will also need to be some language about what happens when a response that contains credentials A, B, and C is sent after receiving individual requests for credentials A, B, C, D and E, and then a new request for credential B is received. Does the first response cause the new request for B to be ignored? Or might a new response containing B, D, and E be sent? Or something else?
```suggestion
            Multiple credentials MAY be present in one such response depending on the capabilities
            of the corresponding [=digital credential/exchange protocol=].
            Each credential SHOULD be present in only one such response, such that a response
            containing multiple credentials is considered to be a response to all individual
            requests for one of the credentials it contains.
```

Reply:
Tweaked to remove the RFC2119
```suggestion
            Multiple credentials might be present in one such response depending on the capabilities
            of the corresponding [=digital credential/exchange protocol=].
            Each credential is meant to be present in only one such response, such that a response
            containing multiple credentials is considered to be a response to all individual
            requests for any of the credentials it contains.
```

Reply:
> ...such that a response containing multiple credentials is considered to be a response to all individual requests for any of the credentials it contains.

@TallTed but the way I see it, the presence of credentials in the response (or lack thereof) is a product of the protocol and requested credential format's capabilities. I'm only trying to highlight that the singular response that should come back should be a response structured for whatever protocol was ultimately leveraged for presentation.

I don't want to try and prescribe here how protocols or doc formats downstream should work, just that the user agent should make sure only one response comes through. The main problem I'm trying to solve with this PR is that the VDC ecosystem will be between OID4VP and Annex C for a while - RPs should be able to specify requests for mDLs using both in a single DC API call to support the widest range of browsers. Only one OID4VP **or** Annex C response should be returned.

Reply:
@MasterKale — As I read [the above](https://github.com/w3c-fedid/digital-credentials/pull/249#discussion_r2304998966), it seems that the intent is a lot like HTTP's content negotiation, where a `GET` request includes an `Accept:` header which contains some number of content types, each with a "quality" ranking which is really a "preference" ranking. The HTTP server has its own set of possible content types, each with its own "quality" ranking. The two sets are melded, winding up with a new "quality" ranked list of media types, and the one with the highest "quality" is generally expected to be returned to the client. The client is not expected to issue multiple `GET` requests to brute-force its way to the first one the server will deliver as requested (and not only because the server may deliver an unrequested media type in response to any `GET`, which may mean that the client receives something it doesn't know how to deal with at all, and depending how that brute-force routine was written, the client might stop there with no useful data *or* it might continue down its list of preferred media types until it actually receives what it asked for).

This is approximately the understanding I had after a FedID call some weeks ago.

I think the text should start by making clear the RP is making a single DC API call which somehow conveys information similar to the HTTP `Accept:` header, in this case including all of the RP's protocol, document type, and ranking preferences. Probably, the permutations of protocol and document type pairings should be treated as the media types, and the ranking preferences remain as ranking; the server should respond to the highest ranked pair that it supports (or do the same calculations as with the HTTP `Accept:` header, and return the resulting highest pair, which may or may not be the highest ranked pair in the RP's requeset).

Note that the HTTP specifications do not mandate "one and only one response to any given HTTP request", but there is no reason why a short sentence saying so couldn't be appended to the HTTP specs, probably as an optional thing that could be switched on or off depending on deployment needs.

I submit that this DC API spec should borrow heavily from [the Content Negotiation section](https://datatracker.ietf.org/doc/html/rfc2295) of the HTTP specs, and add that short "one and only one response" sentence to the resulting DC API spec.